src/java/org/apache/cassandra/metrics/RestorableMeter.java

```diff
@@ -35,9 +36,13 @@
  */
 public class RestorableMeter
 {
-    private static final long TICK_INTERVAL = TimeUnit.SECONDS.toNanos(5);
+    public static final Set<Integer> AVAILABLE_WINDOWS = Set.of(1, 5, 15, 120);
+
+    public static final long TICK_INTERVAL = TimeUnit.SECONDS.toNanos(5);
     private static final double NANOS_PER_SECOND = TimeUnit.SECONDS.toNanos(1);
 
+    private final RestorableEWMA m1Rate;
```

Comment:
Ignore me if it feels like I'm overthinking this, but it feels a bit sad that every user of `RestorableMeter` ends up maintaining all those rate when they generally every use a subset (in the case of the one added by this patch, it uses at most 1, and not even that if the window is 0).

Couldn't it be worth taking a little more time to refactor this class to avoid it. We could have a builder for instance that let use us pick which rate are created on construction. This would also deal a bit more cleanly with the following comment on the "restorable" ctor: the builder could allow to restore some rate and not other.

Note that it's ok for me if we keep the current 4 hardcoded rate internally: we could make this handle arbitrary rates (with, say, some `IntObjectMap`) but keeping the existing  4 fields and simply setting those unused to `null` is fine by me, it's more a matter of cleaning up the API and avoiding waste.

Lastly, if this was me, I'd rename the class too (to maybe something like `MeterWithRates`). Because `Restorable` here just mean there is ctors that take an initial rate, and that feel hardly like the most important thing in the class. But the fact it's called `Restorable` is the only reason for comments like the one here:
https://github.com/datastax/cassandra/pull/1127/files#diff-c57d70c11a1f1373c8e0b486446973fc275ca8bf050218fa6c6af8baecdd3b6fR272

Reply:
Great idea. Pushed c76b6d7 to create only `15m and 120m` rates for sstable read meter and create rate with given window for `partitionIndexReadMeter`.

>  I'd rename the class too (to maybe something like MeterWithRates). Because Restorable here just mean there is ctors that take an initial rate, and that feel hardly like the most important thing in the class.

I am leaving it the same to reduce conflicts from upstream..